pandas/core/series.py

```diff
@@ -435,6 +435,51 @@ def dtypes(self) -> DtypeObj:
 
     @property
     def name(self) -> Label:
+        """
+        Return the name of the Series.
+
+        The name of a Series becomes its index or column name if it is used
+        to form a DataFrame. It is also used whenever displaying the Series
+        using the interpreter, and persists through some transformations.
+
+        Returns
+        -------
+        Label (int, str or other hashable object).
+
+        See Also
+        --------
+        Index.name : Equivalent property for Index object.
+
+        Examples
+        --------
+        >>> s1 = pd.Series([1, 2, 3], dtype=np.int64)
```

Comment:
Nice work @dan1261. I agree with @WillAyd, I think the examples don't represent the common uses of pandas, and that they can somehow confuse users.

What do you think about having two simple examples:
- Create a series with the name as `Series(..., name='whatever')`, and then set to something else
- Create a small dataframe (two columns, three rows), and check the name of one of the columns

I think this will be easier for users to understand the main usage of the `name` attribute. While your examples are good and probably more accurate than this, I think users visiting this page may in most cases appreciate something more basic.

Reply:
Great, I've followed your suggestions, what do you think? I was originally motivated by trying to concat some Series in this manner which led to the example.